src/systems/particle_emitter/ParticleEmitter.cc

```diff
@@ -0,0 +1,186 @@
+    name = _sdf->Get<std::string>("emitter_name");
+  this->dataPtr->emitter.set_name(name);
+
+  // Type. The default type is point.
+  this->dataPtr->emitter.set_type(
+      ignition::msgs::ParticleEmitter_EmitterType_POINT);
+  std::string type = _sdf->Get<std::string>("type", "point").first;
+  if (type == "box")
+  {
+    this->dataPtr->emitter.set_type(
+      ignition::msgs::ParticleEmitter_EmitterType_BOX);
+  }
+  else if (type == "cylinder")
+  {
+    this->dataPtr->emitter.set_type(
+      ignition::msgs::ParticleEmitter_EmitterType_CYLINDER);
+  }
+  else if (type == "ellipsoid")
+  {
+    this->dataPtr->emitter.set_type(
+      ignition::msgs::ParticleEmitter_EmitterType_ELLIPSOID);
+  }
+  else if (type != "point")
+  {
+    ignerr << "Unknown emitter type [" << type << "]. Using [point] instead"
+           << std::endl;
+  }
+
+  // Pose.
+  ignition::math::Pose3d pose =
+    _sdf->Get<ignition::math::Pose3d>("pose");
+  ignition::msgs::Set(this->dataPtr->emitter.mutable_pose(), pose);
+
+  // Size.
+  ignition::math::Vector3d size = ignition::math::Vector3d::One;
+  if (_sdf->HasElement("size"))
+    size = _sdf->Get<ignition::math::Vector3d>("size");
+  ignition::msgs::Set(this->dataPtr->emitter.mutable_size(), size);
+
+  // Rate.
+  this->dataPtr->emitter.set_rate(_sdf->Get<double>("rate", 10).first);
+
+  // Duration.
+  this->dataPtr->emitter.set_duration(_sdf->Get<double>("duration", 0).first);
+
+  // Emitting.
+  this->dataPtr->emitter.set_emitting(_sdf->Get<bool>("emitting", false).first);
+
+  // Particle size.
+  size = ignition::math::Vector3d::One;
+  if (_sdf->HasElement("particle_size"))
+    size = _sdf->Get<ignition::math::Vector3d>("particle_size");
+  ignition::msgs::Set(this->dataPtr->emitter.mutable_particle_size(), size);
+
+  // Lifetime.
+  this->dataPtr->emitter.set_lifetime(_sdf->Get<double>("lifetime", 5).first);
+
+  // Material.
+  if (_sdf->HasElement("material"))
+  {
+    auto materialElem = _sdf->GetElementImpl("material");
+    sdf::Material material;
+    material.Load(materialElem);
+    ignition::msgs::Material materialMsg = convert<msgs::Material>(material);
+  }
+
+  // Min velocity.
+  this->dataPtr->emitter.set_min_velocity(
+    _sdf->Get<double>("min_velocity", 1).first);
+
+  // Max velocity.
+  this->dataPtr->emitter.set_max_velocity(
+    _sdf->Get<double>("max_velocity", 1).first);
+
+  // Color start.
+  ignition::math::Color color = ignition::math::Color::White;
+  if (_sdf->HasElement("color_start"))
+    color = _sdf->Get<ignition::math::Color>("color_start");
+  ignition::msgs::Set(this->dataPtr->emitter.mutable_color_start(), color);
+
+  // Color end.
+  color = ignition::math::Color::White;
+  if (_sdf->HasElement("color_end"))
+    color = _sdf->Get<ignition::math::Color>("color_end");
+  ignition::msgs::Set(this->dataPtr->emitter.mutable_color_end(), color);
+
+  // Scale rate.
+  this->dataPtr->emitter.set_scale_rate(
+    _sdf->Get<double>("scale_rate", 1).first);
```

Comment:
Should we check for invalid values of `scale_rate`? I'm assuming that negative scale values shouldn't be allowed.